app/main/views/applications.py

```diff
@@ -48,6 +56,14 @@ def update_application(application_id):
     if application is None:
         abort(404, "Application '{}' does not exist".format(application_id))
 
+    if application.status == 'submitted' and application_json.get('status') == 'saved':
+        db.session.add(AuditEvent(
+            audit_type=AuditTypes.revert_application,
+            user='',
+            data={},
+            db_object=application
+        ))
+
```

Comment:
One approach would be to add some static methods off the domain classes for these operations, thoughts?